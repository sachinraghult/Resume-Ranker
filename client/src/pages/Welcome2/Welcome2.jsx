import React, { Fragment } from "react";
import { useNavigate } from "react-router-dom";

import styles from "./Welcome2.module.css";
import { User, Url } from "../../enums";

const Welcome2 = () => {
  const navigate = useNavigate();

  const handleSearcher = () => {
    navigate(Url.REGISTER, { state: { type: User.SEARCHER } });
  };

  const handleRecruiter = () => {
    navigate(Url.REGISTER, { state: { type: User.RECRUITER } });
  };

  return (
    <Fragment>
      <div className={styles.loginWrapper}>
        <div className={styles.loginLeft}>
          <h3 className={styles.loginLogo}>Job Finder</h3>
          <span className={styles.loginDesc}>Enter your details...</span>
        </div>
        <div className={styles.welcomeRight}>
          <button
            data-testid="handleSearcher"
            onClick={handleSearcher}
            style={{ backgroundColor: "#4B5D67" }}
          >
            Searcher
          </button>
          <button data-testid="handleRecruiter" onClick={handleRecruiter}>
            Recruiter
          </button>
        </div>
      </div>
      {/* { user.value === 'Register' && } */}
      {/* { user.value === 'Login' && <Login/>} */}
    </Fragment>
  );
};

export default Welcome2;
