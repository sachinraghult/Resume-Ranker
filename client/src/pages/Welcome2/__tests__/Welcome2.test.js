import React from "react";
global.React = React;
import ReactDOM from "react-dom/client";
import Welcome2 from "../Welcome2";
import { render, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import { act } from "react-dom/test-utils";
import { BrowserRouter } from "react-router-dom";

it("renders without crashing", async () => {
  const root = ReactDOM.createRoot(document.createElement("div"));
  await act(async () =>
    render(
      <BrowserRouter>
        <Welcome2 />
      </BrowserRouter>
    )
  );
});

it("renders correctly", () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <Welcome2 />
    </BrowserRouter>
  );
  fireEvent.click(getByTestId("handleSearcher"));
  fireEvent.click(getByTestId("handleRecruiter"));
});

it("matches snapshot", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Welcome2 />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
