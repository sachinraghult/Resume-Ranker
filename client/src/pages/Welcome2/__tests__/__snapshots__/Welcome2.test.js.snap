// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="loginWrapper"
>
  <div
    className="loginLeft"
  >
    <h3
      className="loginLogo"
    >
      Job Finder
    </h3>
    <span
      className="loginDesc"
    >
      Enter your details...
    </span>
  </div>
  <div
    className="welcomeRight"
  >
    <button
      data-testid="handleSearcher"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#4B5D67",
        }
      }
    >
      Searcher
    </button>
    <button
      data-testid="handleRecruiter"
      onClick={[Function]}
    >
      Recruiter
    </button>
  </div>
</div>
`;
